src/pubsub/ua_pubsub_writer.c

```diff
@@ -2006,25 +2038,32 @@ UA_WriterGroup_publishCallback(UA_Server *server, UA_WriterGroup *writerGroup) {
     /* Binary or Json encoding?  */
     if(writerGroup->config.encodingMimeType != UA_PUBSUB_ENCODING_UADP &&
        writerGroup->config.encodingMimeType != UA_PUBSUB_ENCODING_JSON) {
-        UA_LOG_WARNING(&server->config.logger, UA_LOGCATEGORY_SERVER,
+        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER,
                        "Publish failed: Unknown encoding type.");
+        UA_WriterGroup_setPubSubState(server, UA_PUBSUBSTATE_ERROR, writerGroup);
```

Comment:
This change will set the state to Error but the WriterGroup will anyway continue because currently the UA_WriterGroup_setPubSubState only changes the state.

Reply:
You are right. I thought setting the state to error would be better than nothing.
Shall I remove the callback in case of an error?